src/bindings/jna/libelektra-kotlin/src/main/kotlin/org/libelektra/dsl/KeySetDSL.kt

```diff
@@ -0,0 +1,43 @@
+package org.libelektra.dsl
+
+import org.libelektra.Key
+import org.libelektra.KeySet
+
+/**
+ * KeySetDSL provides a DSL API for building KeySets
+ * It provides a way to instantiate complex KeySets in a readable way.
+ * Should not be used directly, but implicitly by calling [keySetOf]
+ *
+ * @param key the key which should be added to the KeySet
+ */
+class KeySetDSL {
+
+    private var keys: MutableSet<Key> = HashSet()
+
+    fun build(): KeySet {
+        return KeySet.create(*keys.toTypedArray())
+    }
+
+    fun key(key: Key) {
+        keys.add(key)
+    }
+
+    fun <T> key(name: String, value: T? = null, vararg metaKeys: Key) {
+        keys.add(
+            Key.create(name, value).apply {
+                metaKeys.forEach {
+                    setMeta(it.name, it.string)
+                }
+            }
+        )
+    }
```

Comment:
```suggestion
    fun key(name: String, initializer: KeyDSL.() -> Unit = {}) {
        keys.add(keyOf(name, initializer))
    }
```

I think that would be more consistent with the `keyOf` DSL function.